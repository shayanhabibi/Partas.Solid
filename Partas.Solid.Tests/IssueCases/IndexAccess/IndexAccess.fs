module Partas.Solid.Tests.IssueCases.IndexAccess

open Partas.Solid
open Fable.Core

[<JS.Pojo>]
type Container (value: Accessor<string>) = member val value: Accessor<string>

[<Erase>]
type TestStringAccess() =
    inherit VoidNode()
    [<Erase>]
    member val index: int = jsNative with get,set
    [<SolidTypeComponent>]
    member props.constructor =
        let context: Container = unbox null
        let character(): string = context.value()[props.index] |> string
        div() {
            character()
        }


