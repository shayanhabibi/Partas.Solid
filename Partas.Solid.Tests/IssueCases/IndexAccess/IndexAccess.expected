import { splitProps } from "solid-js";
import { defaultOf } from "./fable_modules/fable-library-js.5.0.0-alpha.11/Util.js";

export function TestStringAccess(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, ["index"]);
    const context = defaultOf();
    return <div>
        {context.value()[PARTAS_LOCAL.index]}
    </div>;
}